### PR TITLE
added test specific index to indexer

### PR DESCRIPTION
**Pull request by Bento007**
- Added the environment variable `TEST_MODULE_NAME` to test cases that use an elasticsearch index. 

- Modified `get_es_index_name()` to append the `index`  with `TEST_MODULE_NAME` if `DSS_DEPLOYMENT_STAGE == Test`. 

We can now run the test cases in parallel to speed up execution time.

Connects to #355

### CODE REVIEW

**Comment by ttung** [comment]
I'm not sure how I feel about using environment variables to pass data around inside a process.  Can you think of a different way to do this?

**Comment by Bento007** [comment]
I'm open to suggestions. The biggest constraint is I don't want to make an major modifications to the productions code. This was the cleanest way of accomplishing this.

**Comment by Bento007** [comment]
@ttung no longer using environment variables. Please take a look before the sprint end.

File: tests/test_indexer.py [hunk 1]
```diff
@@ -71,6 +72,7 @@ class ESInfo:
     server = None
 
 def setUpModule():
+    IndexSuffix.name = __name__.split('.')[-1]
```
Comment (ttung) [inline tests/test_indexer.py]: you could consider `rsplit(".", 2)` or `rpartition(".")` or you could leave it as is.

File: tests/test_search.py [hunk 2]
```diff
@@ -38,29 +39,30 @@ class ESInfo:
...
         cls.app = dss.create_app().app.test_client()
         with open(os.path.join(os.path.dirname(__file__), "sample_index_doc.json"), "r") as fh:
             cls.index_document = json.load(fh)
 
     def setUp(self):
         dss.Config.set_config(dss.DeploymentStage.TEST)
         self.app = dss.create_app().app.test_client()
-        elasticsearch_delete_index(self.dss_index_name)
-        create_elasticsearch_index(self.dss_index_name, logger)
+        elasticsearch_delete_index(f"*{IndexSuffix.name}")
```
Comment (ttung) [inline tests/test_search.py]: why was the `create_elasticsearch_index(..)` call moved to the specific test?

File: tests/test_search.py [hunk 3]
```diff
@@ -132,9 +134,10 @@ def test_search_returns_N_results_when_N_documents_match_query(self):
         """Create N identical documents. A search query is executed to match the document. All documents created must be
         present in the query results. N is varied across a variety of values.
         """
-        test_matches = [0, 1, 9, 10, 11, 1000, 5000]
+        test_matches = [0, 1, 9, 10, 11, 1000, 10001]
```
Comment (ttung) [inline tests/test_search.py]: why this change?

Comment (Bento007) [reply tests/test_search.py]: normal searching only returns up to the first 10000 results. This check that we can exceed that limit using paging.

File: tests/test_indexer.py (hunk 1 above)
Comment (Bento007) [reply tests/test_indexer.py]: i'll use rsplit, it'll be a little faster

File: tests/test_search.py (hunk 2 above)
Comment (Bento007) [reply tests/test_search.py]: The index is created when the first bundle is indexed. Since no bundles are indexed the index is never created. In all other test cases bundles are indexed before searching.

Comment (ttung) [reply tests/test_search.py]: What's the harm of leaving the create_index call in `setUp`?

Comment (Bento007) [reply tests/test_search.py]: it adds unnecessary steps, otherwise it's harmless.
